tests/components/vultr/test_switch.py

```diff
@@ -19,159 +19,141 @@
-        with patch(
-            "vultr.Vultr.server_list",
-            return_value=json.loads(load_fixture("server_list.json", "vultr")),
-        ), patch("vultr.Vultr.server_start") as mock_start:
-            for device in self.DEVICES:
-                if device.name == "Failed Server":
-                    device.turn_on()
-
-        # Turn on
-        assert mock_start.call_count == 1
-
-    @requests_mock.Mocker()
-    def test_turn_off(self, mock):
-        """Test turning a subscription off."""
-        with patch(
-            "vultr.Vultr.server_list",
-            return_value=json.loads(load_fixture("server_list.json", "vultr")),
-        ), patch("vultr.Vultr.server_halt") as mock_halt:
-            for device in self.DEVICES:
-                if device.name == "A Server":
-                    device.turn_off()
-
-        # Turn off
-        assert mock_halt.call_count == 1
-
-    def test_invalid_switch_config(self):
-        """Test config type failures."""
-        with pytest.raises(vol.Invalid):  # No subscription
-            vultr.PLATFORM_SCHEMA({CONF_PLATFORM: base_vultr.DOMAIN})
-
-    @requests_mock.Mocker()
-    def test_invalid_switches(self, mock):
-        """Test the VultrSwitch fails."""
-        mock.get(
-            "https://api.vultr.com/v1/account/info?api_key=ABCDEFG1234567",
-            text=load_fixture("account_info.json", "vultr"),
-        )
-
-        with patch(
-            "vultr.Vultr.server_list",
-            return_value=json.loads(load_fixture("server_list.json", "vultr")),
-        ):
-            # Setup hub
-            base_vultr.setup(self.hass, VALID_CONFIG)
-
-        bad_conf = {}  # No subscription
-
-        no_subs_setup = vultr.setup_platform(
-            self.hass, bad_conf, self.add_entities, None
-        )
-
-        assert no_subs_setup is not None
-
-        bad_conf = {
-            CONF_NAME: "Missing Server",
-            CONF_SUBSCRIPTION: "665544",
-        }  # Sub not associated with API key (not in server_list)
-
-        wrong_subs_setup = vultr.setup_platform(
-            self.hass, bad_conf, self.add_entities, None
-        )
-
-        assert wrong_subs_setup is not None
+                device.update()
+                device.turn_off()
+
+    # Turn off
+    assert mock_halt.call_count == 1
+
+
+def test_invalid_switch_config():
+    """Test config type failures."""
+    with pytest.raises(vol.Invalid):  # No subscription
+        vultr.PLATFORM_SCHEMA({CONF_PLATFORM: base_vultr.DOMAIN})
+
+
+@pytest.mark.usefixtures("valid_config")
+def test_invalid_switches(hass: HomeAssistant):
+    """Test the VultrSwitch fails."""
+    hass_devices = []
+
+    def add_entities(devices, action):
+        """Mock add devices."""
+        for device in devices:
+            hass_devices.append(device)
+
+    bad_conf = {}  # No subscription
+
+    no_subs_setup = vultr.setup_platform(hass, bad_conf, add_entities, None)
+
+    assert no_subs_setup is not None
```

Comment:
I don't know how much you want to fix in these tests. There's a lot of bad patterns in the existing tests.

The platform setup shouldn't return a boolean at all. On the other hand, we shouldn't test the platform setup directly anyway.

Reply:
I have removed the check on `setup_platform` as this will cause issues when I fix the type hints in the code...
For the rest - I only want the tests to stop failing in partial CI 😄

Reply:
Hi @MartinHjelmare,
Is it ok to merge this as it is? I know the tests are not 100% clean but at least it drops unittest, and it passes partial CI so it's a step in the right direction. Maybe someone else can fix the bad patterns in a follow-up PR later on.

If you do want me to make extra changes then I will need some guidance.

Reply:
This is fine for now. Thanks!